Make hardware compatibility check optional, allowing updates without HW match.

Upstream-Status: Inappropriate No time for upstreaming

--- a/parser/parser.c
+++ b/parser/parser.c
@@ -302,8 +302,8 @@ static int parse_hw_compatibility(parsertype p, void *cfg, struct swupdate_cfg
 
 	setting = find_node(p, cfg, "hardware-compatibility", swcfg);
 	if (setting == NULL) {
-		ERROR("HW compatibility not found");
-		return -1;
+		WARN("HW compatibility not found");
+		return 0;
 	}
 
 	count = get_array_length(p, setting);
--- a/core/hw-compatibility.c
+++ b/core/hw-compatibility.c
@@ -21,6 +21,9 @@ int check_hw_compatibility(struct swupdate_cfg *cfg)
 {
 	struct hw_type *hw;
 	int ret;
+
+	if (LIST_EMPTY(hardware))
+		return 0;
 
 	ret = get_hw_revision(hwt);
 	if (ret < 0)
